lib/sqlalchemy/sql/functions.py

```diff
@@ -33,10 +36,14 @@
 from .selectable import FromClause
 from .selectable import Select
 from .visitors import VisitableType
+from .. import exc as sa_exc
 from .. import util
 
 
+_RegistryList = partial(util.defaultdict, list)
```

Comment:
**mike bayer** (zzzeek) wrote:

I would want this to be a set() or a dict(), so that we don't need to test for containment before calling append() as we do with a list.  I was going to suggest a set() but as I look at the registry logic below I think it might be simpler to make it a dict that actually stores the function objects as well, to ease the transition into case sensitive mode.

Reply:
I didn't want to duplicate the functions so I prefered to just store the names mapping. But you are right, it will be easier to make the future transition. I will update that to use a `dict()`.